app/human-detector-app/components/KeyCloakButton.tsx

```diff
@@ -0,0 +1,49 @@
+import * as React from 'react';
+import { View, StyleSheet, Button } from 'react-native';
+import Constants from 'expo-constants';
+import { useAutoDiscovery } from 'expo-auth-session';
+import KeyCloakInstance from '../auth/KeyCloakAuth';
+
+const keyCloakAuth: KeyCloakInstance = new KeyCloakInstance();
+
+/**
+ * This component will generate a button that will redirect to
+ * KeyCloak authorization.
+ *
+ * @returns KeyCloakButton component
+ */
+export default function KeyCloakButton(): React.ReactElement {
+  const apiUrl: string = Constants.manifest?.extra?.keycloakUrl;
+
+  const discovery = useAutoDiscovery(`${apiUrl}/realms/myrealm`);
+  const [request, response, promptAsync] = keyCloakAuth.getAuthRequest(discovery);
+
+  React.useEffect(() => {
+    if (response?.type === 'success') {
+      const { code } = response.params;
+
+      // Returns authorization code will be connected to a user object
+      keyCloakAuth.exchangeCodeForToken(code, discovery);
```

Comment:
Why are we ignoring the returned token set? Is storing the returned tokens and using our access token as authorization out of scope for this PR?

Reply:
Yeah, im planning on this being in my next PR when I start getting expo tokens sent to the backend.  Basically I just wanted to do all the user stuff in the same PR because that makes more sense to me.